packages/grafana-ui/src/components/Table/HeaderRow.tsx

```diff
@@ -31,17 +32,25 @@ export const HeaderRow = (props: HeaderRowProps) => {
             aria-label={e2eSelectorsTable.header}
             role="row"
           >
-            {headerGroup.headers.map((column: Column, index: number) =>
-              renderHeaderCell(column, tableStyles, showTypeIcons)
-            )}
+            {headerGroup.headers.map((column: Column, index: number) => {
+              if (columnHeaders && columnHeaders.has(index)) {
+                return renderHeaderCell(column, tableStyles, showTypeIcons, columnHeaders.get(index));
+              }
+              return renderHeaderCell(column, tableStyles, showTypeIcons);
+            })}
```

Comment:
You could skip the check and renderHeaderCell() would get `undefined` or the column header. 
```suggestion
            {headerGroup.headers.map((column: Column, index: number) => (
              renderHeaderCell(column, tableStyles, showTypeIcons, columnHeaders.get(index));
            ))}
```